crates/subspace-farmer/src/plot.rs

```diff
@@ -529,6 +578,25 @@ impl IndexHashToOffsetDB {
         }
     }
 
+    fn iter_from<'a>(
+        &'a self,
+        from: &PieceIndexHash,
+    ) -> impl Iterator<Item = io::Result<PieceOffset>> + 'a {
+        let mut iter = self.inner.raw_iterator();
+        iter.seek(&self.get_key(from).to_bytes());
+        std::iter::from_fn(move || match iter.key() {
+            Some(_) => {
+                let value = iter.value().unwrap();
+                let value = <[u8; std::mem::size_of::<PieceOffset>()]>::try_from(value)
+                    .map(PieceOffset::from_le_bytes)
+                    .map_err(|_| io::Error::other("Failed to decode piece offsets from rocksdb"));
```

Comment:
I'd do this instead:
```suggestion
                let value = PieceOffset::from_le_bytes(
                    value
                        .try_into()
                        .expect("Failed to decode piece offsets from rocksdb")
                );
```

Something wen't really bad if we read something from the database, but it is of a wrong size comparing to what we wrote there. This way it is easier to read and I guess return type can be simplified too,